src/player.cpp

```diff
@@ -9610,32 +9610,41 @@ bool player::wield( item& target )
 
     int mv = 0;
 
+    bool target_removed = false;
+    item target_copy = target;
     if( is_armed() ) {
+        inventory old_inv = inv;
```

Comment:
I see. So you're saying that I should be poking around in `player::dispose_item` rather than `player::wield`?

Also, under the current implementation, a player with nothing but a rock (wielded) and jeans (worn) can dispose of the rock in the inventory when wielding the jeans (immediately followed by "Some items tumble to the ground"). I might as well fix that while I'm at it, right?